### DISCUSSION TITLE
Custom endpoint (extension) blocked by CORS

### DISCUSSION THREAD

**Discussion by sapientorius** [directus/directus#15124] [Archive]
### Describe the Bug

I created a custom endpoint (POST) which consumes data and sends an email.

In config I set
```
CORS_ENABLED=true
CORS_ORIGIN=true
```

Now, when I try to use this endpoint when its deployed to my webserver I get the CORS error:
"Access to XMLHttpRequest at 'https://backend.example.domain/contactform' from origin 'https://example.domain' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource."

Regular post request to the api are working. It does only occour on custom endpoints.

### To Reproduce

Set
```
CORS_ENABLED=true
CORS_ORIGIN=true
```
in config

Create a custom endpoint.
Call custom endpoint from another host.

### Errors Shown

"Access to XMLHttpRequest at 'https://backend.example.domain/contactform' from origin 'https://example.domain' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource."

### What version of Directus are you using?

9.14.5 (9.15.1 not possible due to Plesk incompatibility)

### What version of Node.js are you using?

18.7.0

### What database are you using?

MariaDB 10.3.34

### What browser are you using?

Chrome

### How are you deploying Directus?

Plesk

**Comment by rijkvanzanten** [comment]
With a minimal custom endpoint:

```js
// /extensions/endpoints/my-endpoint/index.js

module.exports = (router) => {
	router.get('/', (req, res) => res.send('Hello World'));
};
```

and my env set to

```
# .env
CORS_ENABLED=true
CORS_ORIGIN=true
```

it does seems to work as expected, as a request:

```
GET /my-endpoint/ HTTP/1.1
Origin: example.com
[...]
```

correctly returns 

```
HTTP/1.1 200 OK
Access-Control-Allow-Origin: example.com
[...]

Hello World
```

I'll move this to a discussion as it feels like a configuration / usage problem

**Reply by sapientorius** [reply]
Thank you.

I tried it with a get route instead of a post and now it does work.

Maybe the issue only occours with post routes.

**Reply by rijkvanzanten** [reply]
Still seeing the same behavior on my end with POST as well:

```js
// Endpoint
module.exports = (router) => {
	router.post('/', (req, res) => res.send('Hello World'));
};
```

```
// Request
POST /my-endpoint/ HTTP/1.1
Origin: example.com
[...]

// Response
HTTP/1.1 200 OK
Access-Control-Allow-Origin: example.com
[...]

Hello World
```

**Reply by sapientorius** [reply]
That is strange. I try to figure out, what's the reason on my side.

**Reply by sapientorius** [reply]
It's getting stranger, after changing it back to post, without changing anything else, the post version works as well. 

Sorry for that. Should I notice why this occoured I will post it.
